test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -0,0 +1,143 @@
+    """
+
+    def setUp(self):
+        # delete all schedulers except default
+        self.server.manager(MGR_CMD_DELETE, SCHED, id="@default")
+        TestInterfaces.setUp(self)
+
+    def test_create_successful_scheduler_object(self):
+        """
+        Creates a scheduler object succcessfully .
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+
+    def test_list_schedulers(self):
+        """
+        List a scheduler object.
+        """
+        pass
+
+    def test_delete_schedulers(self):
+        """
+        Delete a scheduler object succcessfully .
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+
+    def test_duplicate_scheduler_name(self):
+        """
+        Check for the scheduler object name.
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        try:
+            rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        except PbsManagerError, e:
+            self.assertTrue(
+                'qmgr: Error (15211) returned from server' in e.msg[1])
+
+    def test_permission_issue_on_sched(self):
+        """
+        Check for the permission to create/delete/modify scheduler object.
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test1",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test2",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_delete_default_sched(self):
+        """
+        Delete default scheduler.
+        """
+        a = {}
+        try:
+            rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="main")
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15214) returned from server' in e.msg[1])
+
+    def test_set_sched_attributes(self):
+        """
+        Set scheduler object attribute.
+        """
+        a = {}
+        # Default values will be set
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        a = {'sched_cycle_length': 1234}
+        rc = self.server.manager(MGR_CMD_SET, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+        # TODO call the list using qmgr and assert the attribute
```

Comment:
I believe what should happen here is that you should add the manager() parameter expect=True.  This will cause manager to set and then query to make sure it is set correctly.  You will need to set sched_cycle_length to something different.